staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -203,8 +203,75 @@ func Jitter(duration time.Duration, maxFactor float64) time.Duration {
 	return wait
 }
 
-// ErrWaitTimeout is returned when the condition exited without success.
-var ErrWaitTimeout = errors.New("timed out waiting for the condition")
+// ErrWaitTimeout is returned when the condition was not satisfied in time.
+//
+// Deprecated: This type will be made private in 1.28 in favor of WaitEndedEarly
+// for checking errors or WrapEndedEarly(err) for returning a typed error.
+var ErrWaitTimeout = ErrorEndedEarly(errors.New("timed out waiting for the condition"))
+
+// EndedEarly returns true if the error returned by Poll or ExponentialBackoff
+// methods indicates the condition was not successful within the method execution.
+// Callers should use this method instead of comparing the error value directly to
+// ErrWaitTimeout, as methods that cancel a context may not return that error.
+//
+// Instead of:
+//
+//	err := wait.Poll(...)
+//	if err == wait.ErrWaitTimeout {
+//	    log.Infof("Wait for operation exceeded")
+//	} else ...
+//
+// Use:
+//
+//	err := wait.Poll(...)
+//	if wait.EndedEarly(err) {
+//	    log.Infof("Wait for operation exceeded")
+//	} else ...
+func EndedEarly(err error) bool {
```

Comment:
❤️

Reply:
name doesn't ring well though

Reply:
I was playing off the `wait.EndedEarly(err)` - still doesn't work for you?  Suggestions?

Reply:
I mean, this error is one of these ones: 
- errErrWaitTimeout
- context.Canceled
- context.DeadlineExceeded

all represent that something was interrupted either because it took too long or it was canceled.

Meanwhile I write this I got it, is "ended earlier than expected", but it may end early for a good thing ... what about `ErrAborted` and `Aborted` or ErrWaitAborted , OpAborted, that emphasize the fact that something was going on and was cancelled

Reply:
Note I'll likely rename errErrWaitTimeout to match whatever this ends up being.  Let me think some more on your suggestions and look at some examples in more recent error handling in the stdlib to see if I can spot some other suggestions.

Reply:
How about `wait.Interrupted(err error) bool` - if the wait was interrupted by something other than the condition (cancel, deadline, or possibly panic)?

Reply:
that sgtm